### PR TITLE
Add support for numpy "_into" methods.

### PR DESCRIPTION
**Pull request by marcoskirsch**
[X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~~[ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~

[X] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Adds a different method rendering template for  methods. In this case, methods that return a waveform as a list of numbers can now take a numpy.ndarray as input and overwrite the contents of the array.

This is tested in NI-FAKE.

Also, enable generation of niScope_FetchBinary8() function using this method.

### List issues fixed by this Pull Request below, if any.

Work towards #511 

### What testing has been done?

Unit tests added to NI-FAKE.
System test added to NI-SCOPE.

### CODE REVIEW

**Comment by marcoskirsch** [comment]
https://travis-ci.org/ni/nimi-python/jobs/314099434

```
E       ImportError: 
E       Importing the multiarray numpy extension module failed.  Most
E       likely you are trying to import a failed build of numpy.
E       If you're working with a numpy git repo, try `git clean -xdf` (removes all
E       files not under version control).  Otherwise reinstall numpy.
E       
E       Original error was: unable to load extension module '/home/travis/build/ni/nimi-python/.tox/64/pypy-test/site-packages/numpy/core/multiarray.pypy-41.so': /home/travis/build/ni/nimi-python/.tox/64/pypy-test/site-packages/numpy/core/multiarray.pypy-41.so: undefined symbol: Py_EnterRecursiveCall
```

Well, that's not good.

File: build/helper/codegen_helper.py [hunk 1]
```diff
@@ -67,7 +71,7 @@ def get_params_snippet(function, parameter_usage_options):
     return ', '.join(snippets)
 
 
-def _get_output_param_return_snippet(output_parameter, parameters, config):
+def _get_output_param_return_snippet(output_parameter, parameters, config, use_numpy_array=False):
```
Comment (texasaggie97) [inline build/helper/codegen_helper.py]: `use_numpy_array` isn't used in this function

File: build/helper/codegen_helper.py [hunk 2]
```diff
@@ -131,108 +137,138 @@ def _get_buffer_parameter_for_size_parameter(parameter, parameters):
...
+            definition = '{0}.{1}({2}.value)  # case 10'.format(module_name, parameter['ctypes_type'], parameter['python_name'])
+        elif corresponding_buffer_parameter is None:
+            definition = '{0}.{1}({2})  # case 9'.format(module_name, parameter['ctypes_type'], parameter['python_name'])
+        elif corresponding_buffer_parameter['direction'] == 'in':
+            # Parameter denotes the size of another (the "corresponding") parameter.
+            definition = '{0}.{1}(0 if {2} is None else len({2}))  # case 6'.format(module_name, parameter['ctypes_type'], corresponding_buffer_parameter['python_name'])
+        else:
+            assert corresponding_buffer_parameter['direction'] == 'out'
+            if corresponding_buffer_parameter['size']['mechanism'] == 'ivi-dance':
+                assert ivi_dance_step == 1 or ivi_dance_step == 2, "ivi-dance has two steps, check metadata for parameter {0}".format(parameter['name'])
```
Comment (texasaggie97) [inline build/helper/codegen_helper.py]: Would it make sense to create an enum for ivi_dance_step to enforce its oneness or twoness?

File: build/helper/codegen_helper.py (hunk 1 above)
Comment (marcoskirsch) [reply build/helper/codegen_helper.py]: Fixed

File: build/helper/codegen_helper.py (hunk 2 above)
Comment (marcoskirsch) [reply build/helper/codegen_helper.py]: Done, helps with code readability.

**Pull request merged**